pages/index.js

```diff
@@ -2,7 +2,14 @@ import Head from 'next/head'
 import Image from 'next/image'
 import styles from '../styles/Home.module.css'
 
-export default function Home() {
+export async function getStaticProps() {
+  const res = await fetch(`${process.env.MOCK_API_HOST}/posts`)
+  const posts = await res.json()
+
+  return { props: { posts } }
+}
```

Comment:
This is the function that gets kicked off when the back-end server calls the webhook. This function then requests all `/posts` and passes it to the React component `Index` as a prop.